### PR TITLE
prepend image tags

### LINKED ISSUE #2769: Multimodal request seems to ignore the prompt at the first request.
If I use any of the existing Ollama examples and try to change the prompt it will not work. It seems like the first prompt is completely ignored - or assumed to be "describe the image".

First I did try the follwing example (with an image from an old Jfokus conference):

```python
import ollama

res = ollama.chat(
        model="llava",
        messages=[
                {
                        'role': 'user',
                        'images': ['../jman-html5.jpg'],
                        'content': 'What is the man wearing?',
                }
        ]
)
```

Running this I always get the default description of the image as if I sent "describe the image".

```
>python image.py
 The image you've provided is a stylized illustration of a character that appears to be a superhero. The character is depicted with a muscular build, wearing a cape with a logo on the chest and the number "5" visible. The cape suggests a classic superhero theme. The character is also wearing a mask with a design that resembles a helmet, which could imply some sort of identity concealment or protection.

In the background, there's a banner with text that says "WWW.MHM.NET," which likely refers to a website related to the fictional superhero team "The Justice League" or similar content. The style of the image is reminiscent of comic book art and is designed in a way that would be appealing for fans of superheroes. 
(base) joakimeriksson@MAPVJLV3F57XY ollama % python image.py
 This is an image of a character, who appears to be a superhero based on the cape and the emblem on the chest. The character has a muscular build, with visible abs and defined biceps, suggesting strength and physical fitness. They are wearing a costume that includes a mask with a design that could be associated with a sports team or a particular brand, as indicated by the logo on the mask. The number "6" is prominently displayed on the chest of the costume, which might signify a team number or an identification number. Additionally, there's text on the image that reads "JWM" and "WWW," which are likely part of the character's name or a branding element. The overall style of the image is reminiscent of comic book art, with bold lines, shading, and a dynamic pose. 

```

Changing to the following works:

```python
import ollama

res = ollama.chat(
        model="llava",
        messages=[
                {
                        'role': 'user',
                        'images': ['../jman-html5.jpg'],
                        'content': ' ',
                },
                {
                        'role': 'user',
                        'content': 'What is the man wearing?',
                        'images': []
                }
        ]
)
```

Running this will give a response about what the man i wearing.
```
>python image.py
 The man in the image is wearing a suit of armor, which appears to be inspired by a superhero theme. 
 He has a muscular build and is standing with his arms crossed. 

```

### LINKED ISSUE #2788: Bug: LLaVA 1.6 34b not respecting initial user prompt
M2 Max MBP 96GB RAM
Ollama 0.1.27
Compared against llama.cpp CLI @b11a93d (same as Ollama version)

Problem:
When using the Ollama CLI or API with an image, the initial text prompt isn't respected. Examples like the one on the [model page](https://ollama.com/library/llava) typically show prompts like "What is in this picture?". However, when changing the initial prompt to something like "Is this image of a llama?" or "How many animals are in this picture?" or even "Ignore the image and tell me the meaning of life", the output is typically a description of the image.

When using the llama.cpp CLI however, these prompts are followed as expected.

When using chat completions in Ollama, a followup question _does_ work properly, so after initially being ignored you can get the expected output on a second attempt.

My suspicion is that this behavior is largely unnoticed, as the default examples are asking for a description and one is being returned. However, this is masking the fact that text prompts used in conjunction with image prompts aren't being properly utilized. As an aside, LM Studio suffers from the same issue.

The below examples use the same image as the one base64 encoded in the above model page CLI example.

Ollama example:
```
❯ ollama run llava:34b-v1.6-q6_K
>>> /set parameter temperature 0.2
Set parameter 'temperature' to '0.2'
>>> How many animals are in this picture? /Users/mike/Downloads/llama.png
Added image '/Users/mike/Downloads/llama.png'
The image you've provided appears to be a cartoon or illustration of an
animal character. It looks like a cute, stylized depiction of a pig with a
happy expression and waving its hand as if saying hello or goodbye. The
art style is simplistic and playful, which is common in many modern
cartoons and emojis.

>>> How many animals are in this picture?
There is only one animal in this picture, which is the cute pig character.
```

llama.cpp example:
```
❯ ./llava-cli -m ../llm-models/cmp-nct/llava-1.6-gguf/ggml-yi-34b-f16-q_5_k.gguf --mmproj ../llm-models/cmp-nct/llava-1.6-gguf/mmproj-llava-34b-f16-q6_k.gguf --image ~/Downloads/llama.png --temp 0.2 -e -p '<|im_start|>system\n<|im_end|><|im_start|>user\n<image>\nHow many animals are in this picture?<|im_end|><|im_start|>assistant\n'
clip_model_load: model name:   vit-large336-custom
clip_model_load: description:  image encoder for LLaVA
clip_model_load: GGUF version: 3
clip_model_load: alignment:    32
clip_model_load: n_tensors:    378
clip_model_load: n_kv:         26
clip_model_load: ftype:        q6_K

...

system_prompt: <|im_start|>system
<|im_end|><|im_start|>user

user_prompt:
How many animals are in this picture?<|im_end|><|im_start|>assistant


There is one animal in this picture, which appears to be a stylized drawing of a pig.

...
```

### PR DESCRIPTION
**Pull request by mxyng**
instead of appending image tags, prepend them which produces better results in general

resolves #2769
resolves #2788

### CODE REVIEW

**Comment by joakimeriksson** [comment]
Will this make the first text prompt to have effects on the first images?
(then this is a very good change - that will improve many many multimodal ollama examples and langchain examples).

Would you like me to test a bit on my multimodal examples? (I have workaround that I can remove)?

**Comment by joakimeriksson** [comment]
Ok - it seems to resolve the issues - I did a quick hack on "business ready detection" using webcamera these two different responses show that it seems to work - this is without my workarounds: 

With your fix:
```
multimodal % python camtest.py
 It appears that the person in the image is not dressed in a formal manner suitable for a business meeting. They are wearing a casual t-shirt and their hair looks unkempt, which does not align with typical business attire. In a professional setting, one would typically wear a neatly pressed shirt, a tie or other form of neckwear, and well-groomed hair. 

```

Without the fix:
```
multimodal % python camtest.py
 In the image, there is a person who appears to be in a room with white walls and a ceiling that looks like it could be made of corrugated metal or cardboard. The individual is dressed in casual attire, wearing a black top, and has short hair that is neatly combed. They are looking directly at the camera with a neutral expression on their face. 
```

Many thanks!

**Comment by joakimeriksson** [comment]
BTW: would you like to have my short piece of code as an ollama example? (I am happy to contribute it if you would like to have another "fun" multimodal app - in the examples repo).

File: server/routes.go [hunk 1]
```diff
@@ -250,6 +250,19 @@ func GenerateHandler(c *gin.Context) {
 		slog.Debug("generate handler", "system", req.System)
 
 		var sb strings.Builder
+		for i := range req.Images {
```
Comment (jmorganca) [inline server/routes.go:253]: should we put this below? (to avoid resetting the string builder)

**Review by jmorganca — APPROVED** [review]
Good to hear re the results improving. I think this is best for now – it seems appending the image tags would "drown out" the text prompt.

**Pull request merged**